addons/website_sale_loyalty/views/website_sale_templates.xml

```diff
@@ -37,18 +37,44 @@
                                         <div class="alert alert-success text-start mt16" role="alert">
                                             <div class="d-flex flex-row">
                                                 <div class="flex-grow-1">
-                                                    <strong t-esc="reward.description"/>
-                                                    <div t-if="reward.program_id.portal_visible">
-                                                        <t t-if="not reward.program_id.is_nominative"><span t-out="website_sale_order._get_real_points_for_coupon(coupon_reward[0])"/> <span t-out="reward.point_name"/></t>
-                                                        <t t-else="">
-                                                            <span>You have <t t-out="website_sale_order._get_real_points_for_coupon(coupon_reward[0])"/></span> <span t-out="reward.point_name"/>
-                                                            <span t-if="reward.program_id.program_type == 'ewallet'"> in your ewallet</span>
-                                                            <t t-if="reward.program_id.program_type != 'ewallet'">
+                                                    <t t-if="reward.program_id.program_type == 'gift_card'">
+                                                        <strong t-esc="reward.description"/><strong> - </strong><strong t-esc="website_sale_order._get_real_points_for_coupon(coupon_reward[0])"/><strong t-esc="reward.point_name"/>
+                                                    </t>
+                                                    <t t-if="reward.program_id.program_type == 'coupons'">
+                                                        <strong>Coupons - </strong><strong t-esc="reward.description"/>
+                                                    </t>
+                                                    <t t-else="">
+                                                        <strong t-esc="reward.description"/>
+                                                    </t>
+                                                    <t t-if="reward.program_id.program_type == 'coupons' or reward.program_id.program_type == 'gift_card'">
```

Comment:
What about `next_order_coupons` ? I don't think it makes sense to rely on the program type to define which display we need to use.
We need to rely on `applies_on` (?unsure actually) and `trigger` on the program instead.